src/core/include/math/hal/intnat-hexl/mubintvecnathexl.h

```diff
@@ -145,15 +133,15 @@ class NativeVectorT final : public lbcrypto::BigVectorInterface<NativeVectorT<In
 
     explicit constexpr NativeVectorT(usint length) noexcept : m_data(length) {}
 
-    constexpr NativeVectorT(usint length, const IntegerType& modulus) noexcept : m_modulus{modulus}, m_data(length) {}
+    constexpr NativeVectorT(usint length, const IntegerType& modulus) noexcept : m_data(length), m_modulus{modulus} {}
 
     constexpr NativeVectorT(usint length, const IntegerType& modulus, const IntegerType& val) noexcept
-        : m_modulus{modulus}, m_data(length, val.Mod(modulus)) {}
```

Comment:
So the intializer list order did not match the declaration order?  Is that's what's going on here?

Had the same problem in another project recently.